code/go/0chain.net/smartcontract/zcnsc/burn.go

```diff
@@ -37,14 +36,13 @@ func (zcn *ZCNSmartContract) Burn(
 	gn, err := GetGlobalNode(ctx)
 	if err != nil {
 		msg := fmt.Sprintf("failed to get global node error: %v, %s", err, info)
+		Logger.Error(msg, zap.Error(err))
 		return "", common.NewError(code, msg)
 	}
 
 	// check burn amount
-	transSAS, err := currency.ParseZCN(float64(trans.Value))
-	if err != nil {
-		return "", err
-	}
+	transSAS := currency.Coin(trans.Value)
```

Comment:
use currency.Int64ToCoin()

Reply:
We will change the txn.Value to uint64 in your currency fix PR, so should be fine leave it as it is.